src/Dfc.CourseDirectory.WebV2/Features/ProviderSearch/ProviderSearch.cshtml

```diff
@@ -0,0 +1,95 @@
+                        <th scope="col" class="govuk-table__header">CD Status</th>
+                        <th scope="col" class="govuk-table__header"></th>
+                    </tr>
+                </thead>
+                <tbody class="govuk-table__body">
+
+                    @if (Model.Providers.Any())
+                    {
+                        foreach (var provider in Model.Providers)
+                        {
+                            <tr class="govuk-table__row" data-testid="search-result-row-@provider.ProviderId">
+                                <td class="govuk-table__cell">@provider.UKPRN</td>
+                                <td class="govuk-table__cell" data-testid="provider-name">@provider.Name</td>
+                                <td class="govuk-table__cell">@provider.Town</td>
+                                <td class="govuk-table__cell">@provider.Postcode</td>
+                                <td class="govuk-table__cell">@provider.ProviderStatus</td>
+                                <td class="govuk-table__cell">@provider.Status</td>
+                                <td class="govuk-table__cell" data-testid="provider-action">
+                                    @if (string.Equals(provider.ProviderStatus, "Active", StringComparison.InvariantCultureIgnoreCase))
+                                    {
+                                        @if (provider.Status == ProviderStatus.Onboarded)
+                                        {
+                                            <a asp-controller="ProviderDashboard" asp-action="Index" asp-route-providerId="@provider.ProviderId" class="govuk-link pttcd-text-nowrap">View dashboard</a>
+                                        }
+                                        else if (provider.Status == ProviderStatus.Registered)
+                                        {
+                                            <form asp-action="OnboardProvider">
+                                                <input type="hidden" name="@nameof(Dfc.CourseDirectory.WebV2.Features.ProviderSearch.OnboardProviderCommand.ProviderId)" value="@provider.ProviderId" />
+                                                <govuk-button class="govuk-!-margin-0 pttcd-text-nowrap">Add provider</govuk-button>
+                                            </form>
+                                        }
+                                    }
+                                </td>
+                            </tr>
+                        }
+                    }
+                    else
+                    {
+                        <tr class="govuk-table__row">
+                            <td class="govuk-table__cell govuk-!-padding-bottom-5 govuk-!-padding-top-5" colspan="7">No providers found</td>
+                        </tr>
+                    }
+
+                </tbody>
+            </table>
+        }
+
+    </div>
+</div>
+
+@section Scripts {
```

Comment:
Might be worth a one line comment here on what this is doing; it took me a while to interpret it.

Reply:
Agreed. It's reasonably dirty!